logging_utilities/django_middlewares/request_middleware.py

```diff
@@ -0,0 +1,115 @@
+import logging
+from typing import Any
+from typing import Callable
+from typing import Optional
+from typing import Type
+from typing import TypeVar
+
+from django.core.handlers.wsgi import WSGIRequest
+
+from logging_utilities.filters.django_append_request import \
+    DjangoAppendRequestFilter
+
+# Create a generic variable that can be 'WrappedRequest', or any subclass.
+T = TypeVar('T', bound='WrappedRequest')
+
+
+class WrappedRequest(WSGIRequest):
+    """WrappedRequest adds the 'logging_filter' field to a standard request to track it.
+    """
+
+    @classmethod
+    def from_parent(
+        cls: Type[T], parent: WSGIRequest, logging_filter: Optional[DjangoAppendRequestFilter]
+    ) -> T:
+        return cls(parent.environ, logging_filter)
+
+    def __init__(self, environ: Any, logging_filter: Optional[DjangoAppendRequestFilter]) -> None:
+        super().__init__(environ)
+        self.logging_filter = logging_filter
+
+
+class AddRequestToLogMiddleware():
+    """Middleware that adds a logging filter *DjangoAppendRequestFilter* to the request.
+    """
+
+    def __init__(self, get_response: Callable[[WSGIRequest], Any], root_logger: str = ""):
+        self.root_logger = root_logger
+        self.get_response = get_response
+
+    def __call__(self, request: WSGIRequest) -> Any:
+        w_request = WrappedRequest.from_parent(request, None)
+        response = self.process_request(w_request)
+        if not response:
+            response = self.get_response(w_request)
+        response = self.process_response(w_request, response)
+
+        return response
+
+    def _find_loggers(self) -> dict[str, logging.Logger]:
+        """Return loggers part of root.
```

Comment:
a little more information about what you do here, i.e. what `root_logger` is about would be helpful. This is not directly obvious also from django middleware doc I think